packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -315,48 +274,58 @@ export function useGridEditRows(apiRef: GridApiRef) {
     [apiRef],
   );
 
-  const handleDoubleClick = React.useCallback(
+  const handleCellEditExit = React.useCallback(
+    (params: GridCellParams, event?: React.SyntheticEvent) => {
+      // TODO check if its propagation was stopped
+      setCellMode(params.id, params.field, 'view');
+
+      // When dispatched by the document, the event is not passed
+      if (!event || !isKeyboardEvent(event)) {
+        return;
+      }
+
+      if (isCellEditCommitKeys(event.key)) {
+        apiRef.current.publishEvent(GRID_CELL_NAVIGATION_KEY_DOWN, params, event);
+        return;
+      }
+      if (isEscapeKey(event.key) || isDeleteKeys(event.key)) {
+        apiRef.current.setCellFocus(params.id, params.field);
+      }
+    },
+    [apiRef, setCellMode],
+  );
+
+  const handleCellDoubleClick = React.useCallback(
     (...args) => {
+      // TODO don't publish if cell is not editable
       apiRef.current.publishEvent(GRID_CELL_EDIT_ENTER, ...args);
     },
     [apiRef],
   );
 
   useGridApiEventHandler(apiRef, GRID_CELL_KEY_DOWN, handleCellKeyDown);
   useGridApiEventHandler(apiRef, GRID_CELL_MOUSE_DOWN, preventTextSelection);
-  useGridApiEventHandler(apiRef, GRID_CELL_DOUBLE_CLICK, handleDoubleClick);
-  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_ENTER, handleEnterEdit);
-  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_EXIT, handleExitEdit);
+  useGridApiEventHandler(apiRef, GRID_CELL_DOUBLE_CLICK, handleCellDoubleClick);
   useGridApiEventHandler(apiRef, GRID_CELL_FOCUS_OUT, handleCellFocusOut);
-  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_PROPS_CHANGE, handleCellEditPropsChange);
   useGridApiEventHandler(apiRef, GRID_COLUMN_HEADER_DRAG_START, handleColumnHeaderDragStart);
-  useGridApiEventHandler(
-    apiRef,
-    GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED,
-    handleCellEditPropsChangeCommited,
-  );
-
-  useGridApiOptionHandler(
-    apiRef,
-    GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED,
-    options.onEditCellChangeCommitted,
-  );
-  // TODO rename to onEditCellPropsChange
-  useGridApiOptionHandler(apiRef, GRID_CELL_EDIT_PROPS_CHANGE, options.onEditCellChange);
-  // TODO remove because onEditCellChange does the same
-  useGridApiOptionHandler(apiRef, GRID_CELL_VALUE_CHANGE, options.onCellValueChange);
+  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_ENTER, handleCellEditEnter);
+  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_EXIT, handleCellEditExit);
+  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_COMMIT, handleCellEditCommit);
+  useGridApiEventHandler(apiRef, GRID_EDIT_CELL_PROPS_CHANGE, handleEditCellPropsChange);
+
+  useGridApiOptionHandler(apiRef, GRID_CELL_EDIT_COMMIT, options.onCellEditCommit);
+  useGridApiOptionHandler(apiRef, GRID_EDIT_CELL_PROPS_CHANGE, options.onEditCellPropsChange);
```

Comment:
I don't get the value proposition of `onEditCellPropsChange`.

```suggestion
  // TODO remove, use onEditRowsModelChange directly
  useGridApiOptionHandler(apiRef, GRID_EDIT_CELL_PROPS_CHANGE, options.onEditCellPropsChange);
```